packages/server/src/http/resolveHTTPResponse.ts

```diff
@@ -351,11 +361,13 @@ export async function resolveHTTPResponse<
       );
       const body = JSON.stringify(transformedJSON);
       unstable_onChunk?.([-1, body]);
+      const { status, headers, ...otherMetadata } = headResponse;
 
       return {
-        status: headResponse.status,
-        headers: headResponse.headers,
+        status,
+        headers,
         body,
+        ...otherMetadata,
       };
```

Comment:
Why do we need to destructure `status` and `headers` from `headResponse` only to re-spread them together in the same object?

Reply:
I did this mostly for readability, but can spread the full `headResponse` if that'd be preferable.